Source/Core/Check.js

```diff
@@ -0,0 +1,144 @@
+/*global define*/
+define([
+    './defaultValue',
+    './defined',
+    './DeveloperError',
+    './isArray'
+    ], function(
+        defaultValue,
+        defined,
+        DeveloperError,
+        isArray) {
+    'use strict';
+
+    /**
+     * Contains functions for checking that supplied arguments are of a specified type
+     * or meet specified conditions
+     * @private
+     *
+     * @exports Check
+     */
+    var Check = {};
```

Comment:
Just to clarify -- you want me to make `Check` a constructor function instead of an object (but with static non-prototype members since `Check` won't need to be instantiated)? I may have misunderstood, I thought it was mostly a matter of documentation so I asked Matt for a good example and got `Transforms` and tried to adhere to that format. The `Transforms` format is not covered by the documentation guide though, so it may have been the wrong example to follow.